### PR TITLE
Mg nav (user story 2)

### PR DESCRIPTION
**Pull request by Gallup93**
Add expect statements to navigation_spec (/home,/cart,/login,/register)

Add /home route to merchant index page, update spec to expect route

Add users controller with login action, Add /login route, add login view

Add /register route, action, and view page

Add User model, add corresponding test, add seeds

Add test for default user nav bar links

### CODE REVIEW

File: spec/features/navigation_spec.rb [hunk 1]
```diff
@@ -32,6 +56,30 @@
         expect(page).to have_content("Cart: 0")
       end
 
+      visit '/home'
```
Comment (muydanny) [inline spec/features/navigation_spec.rb]: this all blew my mind good job

File: spec/features/navigation_spec.rb [hunk 2]
```diff
@@ -32,6 +56,70 @@
...
+
+      visit '/register'
+
+      within 'nav' do
+        expect(page).to have_content("Cart: 0")
+      end
     end
   end
+  # describe "As a Default user" do
+  #   it "I see all links, plus profile and logout links. I do NOT see login or register links" do
```
Comment (sagemlee) [inline spec/features/navigation_spec.rb:79]: This is test is for the next user story, right?

File: db/seeds.rb [hunk 3]
```diff
@@ -19,3 +19,8 @@
 #dog_shop items
 pull_toy = dog_shop.items.create(name: "Pull Toy", description: "Great pull toy!", price: 10, image: "http://lovencaretoys.com/image/cache/dog/tug-toy-dog-pull-9010_2-800x800.jpg", inventory: 32)
 dog_bone = dog_shop.items.create(name: "Dog Bone", description: "They'll love it!", price: 21, image: "https://img.chewy.com/is/image/catalog/54226_MAIN._AC_SL1500_V1534449573_.jpg", active?:false, inventory: 21)
+
+#users with (role: set manually)
+default_1 = User.create(name: "Hank Hill", address: "801 N Alamo St", city: "Arlen", state: "Texas", zip: "61109", email: "ProPAIN@aol.com", password_digest: "W33dWacker", role: 0)
```
Comment (sagemlee) [inline db/seeds.rb:24]: Love these seeds.

File: db/seeds.rb [hunk 4]
```diff
@@ -19,3 +19,8 @@
...
+merchant_1 = User.create(name: "Maude Sloggett", address: "17 Sun Rise St", city: "El Paso", state: "Illinois", zip: "56726", email: "M.Slogget@yahoo.com", password_digest: "Forever27", role: 1)
+admin_1 = User.create(name: "Kurt Cobain", address: "666 Lake Washington Bldv", city: "Seattle", state: "Washington", zip: "32786", email: "GrungeIsDead@gmail.com", password_digest: "Forever27", role: 2)
```
Comment (sagemlee) [inline db/seeds.rb:26]: RIP

File: spec/features/navigation_spec.rb (hunk 2 above)
Comment (Gallup93) [reply spec/features/navigation_spec.rb:79]: correct. User story 3 :)

File: spec/features/navigation_spec.rb [hunk 5]
```diff
@@ -32,6 +56,70 @@
...
+
+      within 'nav' do
+        expect(page).to have_content("Cart: 0")
+      end
+
+      visit '/register'
+
+      within 'nav' do
+        expect(page).to have_content("Cart: 0")
+      end
```
Comment (stellakunzang) [inline spec/features/navigation_spec.rb:75]: Nailed it! That's some solid testing.

File: app/models/user.rb [hunk 6]
```diff
@@ -1,4 +1,16 @@
 
+class User < ApplicationRecord
+  validates_presence_of :name, :address,
+                        :city, :state,
```
Comment (sagemlee) [inline app/models/user.rb:4]: we should add validates presence of role too I think, I might add in a later pull request.

**Review by sagemlee — APPROVED** [review]

**Review by muydanny — APPROVED** [review]

**Pull request merged**